### PR TITLE
Feat/blog cpu vuln 202308

### PR DESCRIPTION
**Pull request by garloff**

### CODE REVIEW

**Comment by garloff** [comment]
I marked it as WIP to indicate that it might not yet be merged as is.
The description of the CPU vulns is done but we lack a few things:
* (minor) prob'ly a few more links, incl. to OS vendor for the fixed kernel/hypervisor (Ubuntu) 
* (minor) advice for users to patch systems under their control: guest kernels, other systems they own
* (major) description from a provider how the rollout is planned and executed. @frosty-geek, @cedegi ,@berendt, @Nils98Ar, @michaelbayr - any input from you to be expected in the next few days? If not, we'll publish without that part...

**Comment by garloff** [comment]
Last open item:
-     (major) description from a provider how the rollout is planned and executed. @frosty-geek, @cedegi ,@berendt, @Nils98Ar, @michaelbayr - any input from you to be expected in the next few days? If not, we'll publish without that part...

File: _i18n/en/_posts/blog/2023-08-29-new-cpu-leaks.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+Some instructions depend on results of previous instructions. The code flow does often
+depend on such results (conditional branches). If the CPU wants to execute these
+instructions despite some previous instruction whose result is needed not yet being
+complete, it can speculate on the outcomes (for example based on previous results).
+CPUs have lots of specialized hardware to do this speculation well: branch predictors,
+return stack buffers, etc.
+
+Nevertheless misspeculation can happen of course. When it does, the CPU needs to
+carefully rewind the state and undo everything that has happened after a mispredicted
+speculation. Results need to be discarded and the state of registers, memory, ...
```
Comment (fkr) [inline _i18n/en/_posts/blog/2023-08-29-new-cpu-leaks.md:34]: Either fill in '...' or remove.

```suggestion
speculation. Results need to be discarded and the state of registers and  memory
```

File: _i18n/en/_posts/blog/2023-08-29-new-cpu-leaks.md [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+be reinstated from before the misprediction. Failures to do so would cause unpredictable
+CPU behaviors and be very serious CPU bugs. CPU vendors are good at avoiding this.
+
+However, even if the official (architectural) state of the CPU is correctly reinstated
```
Comment (fkr) [inline _i18n/en/_posts/blog/2023-08-29-new-cpu-leaks.md:38]:
```suggestion
However, even if the official architectural state of the CPU is correctly reinstated
```

File: _i18n/en/_posts/blog/2023-08-29-new-cpu-leaks.md [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+after misspeculation, there may be traces from the misspeculation, such as internal
+CPU buffers and caches (the microarchitectural state) that have a different state.
```
Comment (fkr) [inline _i18n/en/_posts/blog/2023-08-29-new-cpu-leaks.md:40]:
```suggestion
CPU buffers and caches, the microarchitectural state, that have a different state.
```

File: _i18n/en/_posts/blog/2023-08-29-new-cpu-leaks.md [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+opt-out for these protections.
+
+The stream of vulnerabilities since the original Meltdown and Spectre reports
+has kept CPU vendors and OS developers busy.
+
+One extremely useful tool to know is the
+[spectre-meltdown-checker.sh](https://github.com/speed47/spectre-meltdown-checker).
+When executed (with root privileges) on a host, it checks whether your CPU
+is susceptible to various vulnerabilities and informs you about the mitigation
+status. There's one limitation that you should be aware of, however:
```
Comment (fkr) [inline _i18n/en/_posts/blog/2023-08-29-new-cpu-leaks.md:103]:
```suggestion
status. There's one limitation that you should be aware of:
```

File: _i18n/en/_posts/blog/2023-08-29-new-cpu-leaks.md [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+phantom speculation becomes possible to abuse. This is called
+after the inception movie where ideas are injected during a victim's
+dream. It's described well in their [publication](https://comsec.ethz.ch/research/microarch/inception/).
+The researchers have been able to create a flow where a harmless
+`XOR` instruction turns into a misspeculation to become a recursive call,
+overflowing the return stack buffer (RSB) with an attacker controlled
+value that will be speculatively jumped to upon the next `RET` instruction.
+The alternative name Speculative Return Stack Overflow (SRSO) is thus used
+as well. Inception was published on Aug 8 and has been assigned
+CVE-2023-20569.
```
Comment (fkr) [inline _i18n/en/_posts/blog/2023-08-29-new-cpu-leaks.md:161]:
```suggestion
[CVE-2023-20569](https://cve.mitre.org/cgi-bin/cvename.cgi?name=2023-20569).
```

File: _i18n/en/_posts/blog/2023-08-29-new-cpu-leaks.md [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+notices that its dreaming and cleans up on awaking), so that the
+phantom speculation becomes possible to abuse. This is called
+after the inception movie where ideas are injected during a victim's
+dream. It's described well in their [publication](https://comsec.ethz.ch/research/microarch/inception/).
+The researchers have been able to create a flow where a harmless
+`XOR` instruction turns into a misspeculation to become a recursive call,
+overflowing the return stack buffer (RSB) with an attacker controlled
+value that will be speculatively jumped to upon the next `RET` instruction.
+The alternative name Speculative Return Stack Overflow (SRSO) is thus used
+as well. Inception was published on Aug 8 and has been assigned
```
Comment (fkr) [inline _i18n/en/_posts/blog/2023-08-29-new-cpu-leaks.md:160]:
```suggestion
as well. Inception was published on August 8 and has been assigned
```

File: _i18n/en/_posts/blog/2023-08-29-new-cpu-leaks.md [hunk 7]
```diff
@@ -0,0 +1,335 @@
+---
+layout: post
+title: "New CPU leaks (Aug 2023)"
```
Comment (fkr) [inline _i18n/en/_posts/blog/2023-08-29-new-cpu-leaks.md:3]:
```suggestion
title: "New CPU leaks (August 2023)"
```

**Review by fkr — APPROVED** [review]
Looks very good to me. For my taste it is too long, but that is personal preference. I tried to refrain from commenting to many stylistic items. 
Aside from the inline comments (feel free to ignore / discard them), there is one item I do miss:

The blog post is meant for the SCS blog. However SCS is not mentioned until the very end. Imho there should be an introduction that briefly in 1 or 2 sentences states, why this topic is relevant for the scs community.

**Comment by frosty-geek** [comment]
Hi Kurt,

> * (major) description from a provider how the rollout is planned and executed. @frosty-geek, @cedegi ,@berendt, @Nils98Ar, @michaelbayr - any input from you to be expected in the next few days? If not, we'll publish without that part...

my 2cents are to push it out. 

Ralf

**Pull request merged**